src/Jackett.Common/Indexers/Milkie.cs

```diff
@@ -0,0 +1,167 @@
+                link: "https://milkie.cc/",
+                caps: new TorznabCapabilities(),
+                configService: configService,
+                client: wc,
+                logger: l,
+                p: ps,
+                configData: new ConfigurationDataAPIKey())
+        {
+            Encoding = Encoding.UTF8;
+            Language = "en-us";
+            Type = "private";
+
+            AddCategoryMapping("1", TorznabCatType.Movies, "Movies");
+            AddCategoryMapping("2", TorznabCatType.TV, "TV");
+            AddCategoryMapping("3", TorznabCatType.Audio, "Music");
+            AddCategoryMapping("4", TorznabCatType.PCGames, "Game");
+            AddCategoryMapping("4", TorznabCatType.Console, "Game");
+            AddCategoryMapping("5", TorznabCatType.Books, "Ebook");
+            AddCategoryMapping("6", TorznabCatType.PC0day, "App");
+            AddCategoryMapping("6", TorznabCatType.PCISO, "App");
+            AddCategoryMapping("6", TorznabCatType.PCMac, "App");
+            AddCategoryMapping("6", TorznabCatType.PCPhoneAndroid, "App");
+            AddCategoryMapping("6", TorznabCatType.PCPhoneIOS, "App");
+            AddCategoryMapping("6", TorznabCatType.PCPhoneOther, "App");
+            AddCategoryMapping("6", TorznabCatType.OtherMisc, "App");
+            AddCategoryMapping("7", TorznabCatType.XXX, "Adult");
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            try
+            {
+                var results = await PerformQuery(new TorznabQuery());
+
+                if (results.Count() == 0)
+                    throw new Exception("Testing returned no results!");
+
+                IsConfigured = true;
+                SaveConfig();
+                return IndexerConfigurationStatus.Completed;
+            }
+            catch (Exception e)
+            {
+                IsConfigured = false;
+                throw new ExceptionWithConfigData(e.Message, configData);
+            }
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var queryParams = new NameValueCollection
+            {
+                { "ps", "100" }
+            };
+
+            if (!string.IsNullOrWhiteSpace(query.SearchTerm))
+            {
+                queryParams.Add("query", query.SearchTerm);
+            }
+
+            if (query.HasSpecifiedCategories)
+            {
+                queryParams.Add("categories", string.Join(",", MapTorznabCapsToTrackers(query)));
+            }
+
+            var endpoint = TorrentsEndpoint + "?" + queryParams.GetQueryString();
+            var headers = new Dictionary<string, string>() { { "x-milkie-auth", configData.Key.Value } };
+            var jsonResponse = await RequestStringWithCookies(endpoint, null, null, headers);
```

Comment:
Use named parameters to skip repeating default parameters.
```suggestion
            var jsonResponse = await RequestStringWithCookies(endpoint, headers: headers);
```